packages/compiler/src/core/program.ts

```diff
@@ -248,27 +282,24 @@ export async function compile(
   program.checker.checkProgram();
 
   if (!continueToNextStage) {
-    return program;
+    return { program, shouldEmit: false };
   }
+
   // onValidate stage
   await runValidators();
 
   validateRequiredImports();
 
   await validateLoadedLibraries();
+
   if (!continueToNextStage) {
-    return program;
+    return { program, shouldEmit: false };
   }
 
   // Linter stage
   program.reportDiagnostics(linter.lint());
 
-  // Emitter stage
-  for (const instance of emitters) {
-    await runEmitter(instance);
-  }
-
-  return program;
+  return { program, shouldEmit: true };
```

Comment:
Can we honor `--no-emit` option, if `--no-emit` : true, we will return `shouldEmit: false` to skip run emitter?

When we define in emitters in tspconfig.yaml, even we flag `--no-emit`, the typespec/compile will run emitter to emit code, the workaround is that each emitter honor `--no-emit` flag. This approach is not acceptable for customized emitter in future.

Reply:
this is out of scope of this pr, there is a separate discussion for this I hope to resolve before 1.0-rc https://github.com/microsoft/typespec/issues/401